sql.go

```diff
@@ -21,18 +21,32 @@ type DBDriver interface {
 type Filter struct {
 	Glue      string        `json:"glue"`
 	Field     string        `json:"field"`
-	Condition Condition     `json:"condition"`
+	Type      string        `json:"type"`
+	Predicate string        `json:"predicate"`
+	Filter    string        `json:"filter"`
+	Value     interface{}   `json:"value"`
 	Includes  []interface{} `json:"includes"`
-	Kids      []Filter      `json:"rules"`
+	Rules     []Filter      `json:"rules"`
+}
+
+func (f *Filter) getValues() []interface{} {
+	valueMap, ok := f.Value.(map[string]interface{})
+	if !ok {
+		return []interface{}{f.Value}
+	}
+
+	return []interface{}{valueMap["start"], valueMap["end"]}
 }
 
 type CustomOperation func(string, string, []interface{}) (string, []interface{}, error)
+type CustomPredicate func(string, string, []interface{}) (string, []interface{}, error)
```

Comment:
Its more complicated than necessary

`type CustomPredicate func(string, string) (string, error)`

predicates don't have any parameters, it's basically a name of an operator which need to be applied to a field before placing it in a sql operation